### PR TITLE
Us 3 user navigation

### PR DESCRIPTION
**Pull request by the-color-bliu**
- [x] done

User Story 3, User Navigation

As a default user
I see the same links as a visitor
Plus the following links
- [x] a link to my profile page ("/profile")
- [x] a link to log out ("/logout")

Minus the following links
- [x] I do not see a link to log in or register

- [x] I also see text that says "Logged in as Mike Dao" (or whatever my name is)

### CODE REVIEW

File: app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -10,12 +10,21 @@
...
+      <%= link_to "Home", "/"%>
       <%= link_to "All Merchants", "/merchants"%>
       <%= link_to "All Items", "/items"%>
-      <%= link_to "Cart: #{cart.total_items}", "/cart" %>
-      <%= link_to "Home", "/"%>
-      <%= link_to "Log In", "/login"%>
-      <%= link_to "Register", "/register"%>
+      <section class="right-pane">
+        <%= link_to "Cart: #{cart.total_items}", "/cart" %>
+      </section>
```
Comment (helloeduardo) [inline app/views/layouts/application.html.erb:18]: Nice job getting it to align to the right! I added an issue #6 for the rest of the links.

File: app/assets/stylesheets/application.css [hunk 2]
```diff
@@ -90,6 +90,15 @@ section[id*='review'] {
    font-size: 17px;
  }
```
Comment (BJSherman80) [inline app/assets/stylesheets/application.css:92]: I like how you put the cart on the right side of the nav

File: spec/features/navigation_spec.rb [hunk 3]
```diff
@@ -84,4 +84,33 @@
...
+        expect(page).to have_content("Home")
+        expect(page).to have_content("Cart: 0")
+        expect(page).to have_content("All Merchants")
+        expect(page).to have_content("All Items")
+        expect(page).to have_content("Profile")
+        expect(page).to have_content("Log Out")
+        expect(page).not_to have_content("Log In")
+        expect(page).to have_no_content("Register")
+        expect(page).to have_content('Logged in as Batman')
+        save_and_open_page
```
Comment (helloeduardo) [inline spec/features/navigation_spec.rb:112]: Gotta remove the save_and_open_page

File: app/controllers/application_controller.rb [hunk 4]
```diff
@@ -11,4 +11,8 @@ def current_user
     @current_user ||= User.find(session[:user_id]) if session[:user_id]
   end
 
+  def current_default_user?
+    current_user && current_user.default_user?
+  end
+
```
Comment (BJSherman80) [inline app/controllers/application_controller.rb:17]: great job, nice way to authenticate if this is a default user

**Review by helloeduardo — APPROVED** [review]
Looks good!

File: app/models/user.rb [hunk 5]
```diff
@@ -5,6 +5,8 @@ class User < ApplicationRecord
   validates :password, confirmation: { case_sensitive: true }
   validates_presence_of :name, :address, :city, :state, :zip
 
+  enum role: [:default_user, :merchant_employee, :admin_user]
+
```
Comment (BJSherman80) [inline app/models/user.rb:9]: Just note the enum can also be a hash with assigned values like  
```
enum role: {default_user: 0, etc...}
```

File: db/migrate/20201028014202_add_role_to_users.rb [hunk 6]
```diff
@@ -0,0 +1,5 @@
+class AddRoleToUsers < ActiveRecord::Migration[5.2]
+  def change
+    add_column :users, :role, :integer, default: 0
+  end
+end
```
Comment (BJSherman80) [inline db/migrate/20201028014202_add_role_to_users.rb:5]: nice job setting the default here!

**Review by BJSherman80 — APPROVED** [review]

**Pull request merged**